docs/source/python/api/datatypes.rst

```diff
@@ -72,6 +72,12 @@ These should be used to create Arrow data types and schemas.
    field
    schema
    from_numpy_dtype
+   bool8
+   uuid
+   json_
+   union
+   dense_union
+   sparse_union
```

Comment:
I think a better order for these is:
- `union`, `dense_union`, and `sparse_union` just above `opaque`
- `bool8`, `uuid`, and `json_` just below `opaque`